### ISSUE TITLE
importlib not follow dynamic imports

### ISSUE THREAD

**Issue by Ja7adR** [Nuitka/Nuitka#1017]
```
0.6.12.3
Python: 3.7.8 (tags/v3.7.8:4b47a5b6ba, Jun 28 2020, 07:55:33) [MSC v.1916 32 bit (Intel)]
Executable: D:\Project\Python\MoeinBackup\.venv\Scripts\python.exe
OS: Windows
Arch: x86
```

I implement dynamic import.

sample implement code : https://github.com/MoeinSoftwares/importlib_test

```python
import importlib
from src import settings


class Runner(object):
    def __init__(self, *args, **kwargs):
        self.args = args
        self.kwargs = kwargs

    def run_modules(self):
        for m in settings.modules:
            module = importlib.import_module(settings.modulesdir + '.' + m)
            cls = getattr(module, m.title())()
            cls.run() 
```

Nuitka flags : `nuitka --standalone --show-progress --follow-imports index.py`

But after compile and run executable file I get this error :

```
Traceback (most recent call last):
  File "D:\Project\Python\IMPORT~1\INDEX~1.DIS\index.py", line 6, in <module>
  File "D:\Project\Python\IMPORT~1\INDEX~1.DIS\src\__init__.py", line 12, in run_modules
  File "importlib.py", line 127, in import_module
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 953, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 965, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'src.modules'
```

In[ approach 2](https://github.com/MoeinSoftwares/importlib_test/tree/approach2) I use `__import__` and same error :

```
Traceback (most recent call last):
  File "D:\Project\Python\IMPORT~1\INDEX~1.DIS\index.py", line 6, in <module>
  File "D:\Project\Python\IMPORT~1\INDEX~1.DIS\src\__init__.py", line 12, in run_modules
ModuleNotFoundError: No module named 'src.modules'
```

**Comment by kayhayen** [comment]
You can use `--include-module` on the module you expect to happen there. If you are lazy, you can also use `--include-package` on whole packages, but that can be too much code included. 

At this time, core Nuitka doesn't support run time profiling of imported modules, but I expect it to arrive later this year along with detection of used DLLs. Until then, you get to do this manually.

**Comment by ralienpp** [comment]
I suggest a small usability improvement: Nuitka could _hint the user_ that dynamic imports were encountered, and that they should pay attention to line X in file Y.

The hint might do the following:
- Give a heads-up that the program might not work.
- Say in what file:line the dynamic imports occur.
- Urge the user to explicitly add those imports via `--include-package` or in some other way.

Without a hint, the user will observe that the program doesn't work, but have little clue about potential root causes. I'm dealing with a very large project written by other people, I was unaware that such dynamic imports are there in the first place. So I spent a lot of time figuring it out.

**Comment by kayhayen** [comment]
That small usability improvement was how I implemented it from the outset @ralienpp and that didn't help (me) a single bit.

Rather, people who didn't have the issue, never understood what to do about the warning. It's 3rd part code, and many didn't even get what it could be about. Compare that to the amount of people who understand `ImportError` at run time. So they asked me. And then I got plenty of reports with the warning being ignored.

The midterm plan for this is to handle `ImportError` in some sort of debug mode that needs to be switched to release mode, telling them why a module is not included.
